proxy_network.c

```diff
@@ -856,7 +856,7 @@ static void proxy_event_handler(evutil_socket_t fd, short which, void *arg) {
                 _reset_bad_backend(be, P_BE_FAIL_WRITING);
                 _backend_failed(be);
             } else {
-                flags = be->can_write ? EV_READ|EV_TIMEOUT : EV_READ|EV_WRITE|EV_TIMEOUT;
+                flags |= EV_TIMEOUT;
```

Comment:
For my understand, this corresponds to your first point, so that now we use the `flags` returned by `_flush_pending_write()` and only append `timeout`.

correct me: it appears to me that the previous logic is still valid?

trivial: let `_flush_pending_write()` return flags already contains timeout event.

Reply:
`_flush_pending_write` is called in two places; the other place uses the flags directly, this one ignored it and re-sets them, all callers should match logic to reduce potential errors.

Not sure changing the EV_TIMEOUT bit is anything but a style change.